pkg/resources/shared_database_acceptance_test.go

```diff
@@ -25,6 +25,13 @@ func TestAcc_CreateSharedDatabase_Basic(t *testing.T) {
 	newId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
 	newComment := random.Comment()
 
+	t.Cleanup(func() {
+		// TODO(SNOW-1562172: Create a better solution for this type of situations
+		// It's needed, because we have to wait for Snowflake to "register" the cleanup after this test.
+		// Without it, the next test would fail.
+		time.Sleep(time.Second * 60)
```

Comment:
It ended up as a pretty big number (I tried a bunch of lower ones), lmk what do you think we should do about it (or just leave it as is for now)

Reply:
BTW there are also functions like [require.Eventually](https://pkg.go.dev/github.com/stretchr/testify/require#Eventually).

Reply:
Managed to change it with the use of require.Eventually and periodic checking with 1 minute timeout